src/org/labkey/test/tests/DomainDesignerTest.java

```diff
@@ -1498,6 +1507,172 @@ public void addUpdateRemoveConditionalFormat() throws Exception
+        File roundTripFile = sampleTypeDesignerPage.fieldsPanel().clickExportFields();
+        List<PropertyDescriptor> roundTrippedFields = getFieldsFromExportFile(roundTripFile);
+
+        for (PropertyDescriptor intendedField : dgen.getColumns())
+        {
+            String fieldName = intendedField.getName();
+            if (!fieldName.equals("name"))  // don't expect name col to export from sampletype, it is special
+            {
+                PropertyDescriptor exportedField = roundTrippedFields.stream().filter(a -> a.getName().equals(fieldName))
+                        .findFirst().orElse(null);
+                assertNotNull("expected field [" + intendedField.getName() + "] was not exported", exportedField);
+                verifyExpectedFieldProperties(intendedField, exportedField);
+            }
+        }
+    }
+
+    private void verifyExpectedFieldProperties(PropertyDescriptor intendedField,  PropertyDescriptor actualField)
+    {
+        log("verifying properties for field [" +intendedField.getName()+ "]");
+        assertThat(actualField.getDescription(), is(intendedField.getDescription()));
+        assertThat(actualField.getLabel(), is(intendedField.getLabel()));
+        assertThat(actualField.getRequired(), is(intendedField.getRequired()));
+        assertThat(actualField.getHidden(), is(intendedField.getHidden()));
+        assertThat(actualField.getMvEnabled(), is(intendedField.getMvEnabled()));
+        assertThat(actualField.getDimension(), is(intendedField.getDimension()));
+        if (intendedField.getPHI() != null)
+            assertThat(actualField.getPHI(), is(intendedField.getPHI()));
+        assertThat(actualField.getMeasure(), is(intendedField.getMeasure()));
+        assertThat(actualField.getFormat(), is(intendedField.getFormat()));
+        for (ConditionalFormat intendedFormat : intendedField.getConditionalFormats())
+        {
+            ConditionalFormat actualFormat = intendedField.getConditionalFormats().stream()
+                    .filter(a-> a.toJSON().equals(intendedFormat.toJSON())).findFirst().orElse(null);
+            assertNotNull("conditional formats did not export with full fidelity", actualFormat);
+        }
+        // would like to do validators, but this part of the remoteAPI is incomplete; validators are settable
+        // on FieldDefinition, but not gettable on PropertyDescriptor
+    }
+
+    private List<PropertyDescriptor> getFieldsFromExportFile(File exportFile) throws Exception
+    {
+        JSONParser parser = new JSONParser();
+        JSONArray  jsonArray = (JSONArray) parser.parse(new FileReader(exportFile));
+
+        List<PropertyDescriptor> exportFields = new ArrayList<>();
+        for (int i=0; i < jsonArray.size(); i++)
+        {
+            exportFields.add(new PropertyDescriptor((JSONObject) jsonArray.get(i)));
+        }
+
+        return exportFields;
+    }
```

Comment:
Have this return a `Map<String, PropertyDescriptor>`. The `fields.stream().filter(a-> a.getName().equals("intField"))...` pattern is a bit cumbersome and we have the field names readily available here.
```suggestion
    private Map<String, PropertyDescriptor> getFieldsFromExportFile(File exportFile) throws Exception
    {
        JSONParser parser = new JSONParser();
        JSONArray  jsonArray = (JSONArray) parser.parse(new FileReader(exportFile));

        Map<String, PropertyDescriptor> exportFields = new HashMap<>();
        for (int i=0; i < jsonArray.size(); i++)
        {
            PropertyDescriptor pd = new PropertyDescriptor((JSONObject) jsonArray.get(i));
            exportFields.put(pd.getName(), pd);
        }

        return exportFields;
    }
```

Reply:
OK, I'll do this.
Unfortunately, this breaks my previous approach of ensuring that field ordering survives export/round-trip unless I use a map type that ensures key ordering